simple_parsing/replace.py

```diff
@@ -90,3 +105,125 @@ def replace(obj: DataclassT, changes_dict: dict[str, Any] | None = None, **chang
     replace_kwargs.update(changes)
 
     return dataclasses.replace(obj, **replace_kwargs)
+
+
+def unflatten_selection_dict(
+    flattened: Mapping[str, V], keyword: str = "__key__", sep: str =".", recursive: bool = False
+) -> PossiblyNestedDict[str, V]:
+    """
+    This function convert a flattened dict into a nested dict
+    and it inserts the `keyword` as the selection into the nested dict.
+
+    >>> unflatten_selection_dict({'ab_or_cd': 'cd', 'ab_or_cd.c_or_d': 'd'})
+    {'ab_or_cd': {'__key__': 'cd', 'c_or_d': 'd'}}
+    
+    >>> unflatten_selection_dict({'lv1': 'a', 'lv1.lv2': 'b', 'lv1.lv2.lv3': 'c'})
+    {'lv1': {'__key__': 'a', 'lv2': 'b', 'lv2.lv3': 'c'}}
```

Comment:
I would suggest making `recursive=True` by default. The non-recursive case seems a bit strange to me.
Also, is this function expected to be used by users? If not, I'd prefix it with an underscore (`_unflatten_selection_dict`) to signal that it is an internal function only used in this module.
I would also place it below other "public" functions like `replace` and `replace_subgroups`.

Reply:
Thanks @lebrice, I have made changes for all your suggestions.
But the test failed due to the Literal issue #259. Once #260 merged, this PR will pass the test.